### PR TITLE
8.0.x

### PR DESCRIPTION
**Pull request by afdw**
3D book, new forge version and auto workbenches

### CODE REVIEW

File: common/buildcraft/api/bpt/BlueprintAPI.java [hunk 1]
```diff
@@ -94,7 +94,8 @@ public static IBptAction deserializeAction(NBTTagCompound nbt) {
         NBTTagCompound data = nbt.getCompoundTag("data");
         IBptReader<IBptAction> reader = getActionDeserializer(new ResourceLocation(id));
         if (reader != null) {
-            return reader.deserialize(data);
+//            return reader.deserialize(data); // FIXME: Anton
+            return null; // FIXME: Anton
```
Comment (AlexIIL) [inline common/buildcraft/api/bpt/BlueprintAPI.java]: I've just fixed this in a commit.

**Comment by AlexIIL** [comment]
This is interesting, I didn't think I would get help for this!

Although a few things have (just) changed because of a commit I just made.

Because BC is undergoing a relicense you would have to agree to it (here)[https://github.com/BuildCraft/BuildCraft/issues/3004] before I can accept this. Also all of the "buildcraft.api" packages are going to be relicensed to MIT and put in the official API repository when its settled down, so you would also need to agree to them moving and being under MIT rather than MPL2.

**Comment by AEnterprise** [comment]
@AlexIIL due to the new BC code license i sugest adding a CLA to BC, github has integration for it that can do automatic checking and indicating if the contributors in a PR signed or not

File: common/buildcraft/api/bpt/BlueprintAPI.java [hunk 2]
```diff
@@ -12,7 +12,7 @@
 import net.minecraft.util.ResourceLocation;
 import net.minecraft.util.StringUtils;
 
-import buildcraft.api.IUniqueReader;
+//import buildcraft.api.IUniqueReader;
```
Comment (AEnterprise) [inline common/buildcraft/api/bpt/BlueprintAPI.java]: i would advise agains commenting out imports, it's easy to have your IDE auto-re-import but the comment will stay there being useless

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,6 +1,6 @@
-#Sat May 21 15:16:57 BST 2016
+#Tue Jun 28 17:10:16 MSK 2016
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.13-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-2.13-all.zip
```
Comment (AEnterprise) [inline gradle/wrapper/gradle-wrapper.properties]: do we want to use the "all" version? the bin should be enough and it's just a larger download that isn't needed i think

**Comment by AlexIIL** [comment]
Seems good, thanks :)

**Comment by AlexIIL** [comment]
So I've just tested the new Guide Opening animation you made. Its awesome :D

**Comment by LuigiHutch** [comment]
Gif plz :)

**Comment by afdw** [comment]
@AlexIIL Is there any place (chat, mail, etc) we can talk?

**Comment by AlexIIL** [comment]
@afdw The IRC channel "buildcraft" (espernet) is probably the best place, however you can always PM me on there if it has to be private.

**Pull request merged**